build_web_compilers/lib/src/web_entrypoint_builder.dart

```diff
@@ -89,69 +95,152 @@ class WebEntrypointBuilder implements Builder {
-    var compiler = WebCompiler.fromOptionName(compilerOption);
-
-    return WebEntrypointBuilder(
-      compiler,
-      dart2JsArgs: _parseCompilerOptions(options, _dart2jsArgsOption),
-      dart2WasmArgs: _parseCompilerOptions(options, _dart2wasmArgsOption),
-      nativeNullAssertions:
-          options.config[_nativeNullAssertionsOption] as bool?,
+  factory EntrypointBuilderOptions.fromOptions(BuilderOptions options) {
+    const deprecatedOptions = [
+      'enable_sync_async',
+      'ignore_cast_failures',
+    ];
+
+    const compilerOption = 'compiler';
+    const compilersOption = 'compilers';
+    const dart2jsArgsOption = 'dart2js_args';
+    const dart2wasmArgsOption = 'dart2wasm_args';
+    const nativeNullAssertionsOption = 'native_null_assertions';
+    const loaderOption = 'loader';
+    String? defaultLoaderOption;
+
+    const supportedOptions = [
+      compilersOption,
+      compilerOption,
+      dart2jsArgsOption,
+      nativeNullAssertionsOption,
+      dart2wasmArgsOption,
+      loaderOption,
+    ];
+
+    var config = options.config;
+    var nativeNullAssertions =
+        options.config[nativeNullAssertionsOption] as bool?;
+    var compilers = <EnabledEntrypointCompiler>[];
+
+    validateOptions(config, supportedOptions, 'build_web_compilers:entrypoint',
+        deprecatedOptions: deprecatedOptions);
+
+    // The compilers option is a map of compiler names to options only applying
+    // to that compiler, which allows compiling with multiple compilers (e.g.
+    // dart2js + dart2wasm). Since the default builder configuration doesn't
+    // use the compilers key, we preserve backwards compatibility.
+    if (config.containsKey(compilersOption)) {
+      var configuredCompilers = (config[compilersOption] as Map?)
+              ?.cast<String, Map<String, Object?>?>() ??
+          const {};
+      var hasDart2Wasm = false;
+
+      for (var MapEntry(:key, :value) in configuredCompilers.entries) {
+        const extensionOption = 'extension';
+        const argsOption = 'args';
+        const supportedOptions = [extensionOption, argsOption];
+        validateOptions(value ?? const {}, supportedOptions,
+            'build_web_compilers:entrypoint');
+
+        var compiler = WebCompiler.fromOptionName(key);
+        compilers.add(EnabledEntrypointCompiler(
+          compiler: compiler,
+          extension: value?[extensionOption] as String? ??
+              (configuredCompilers.length == 1
+                  ? compiler.entrypointExtensionWhenOnlyCompiler
+                  : compiler.entrypointExtension),
+          compilerArguments: _parseCompilerOptions(
+              value?[argsOption], '$compilersOption.$key'),
+        ));
+
+        hasDart2Wasm |= compiler == WebCompiler.Dart2Wasm;
+      }
+
+      if (hasDart2Wasm) {
+        defaultLoaderOption = '.dart.js';
+      }
+    } else {
+      var compilerName = config[compilerOption] as String? ?? 'dartdevc';
```

Comment:
I am a bit surprised that we have a default of `dartdevc` here, we shouldn't need that I don't think? Can we ever get here without any compiler being set, given the default options?

Reply:
This caught be by surprise as well, but the `?? 'dartdevc'` has been there before and there is no `compiler: dartdevc` in `build.yaml` under `dev_options`.

Should I remove the fallback from here and add a default to `build.yaml`?

Reply:
> and there is no `compiler: dartdevc` in `build.yaml` under `dev_options`.

Oh ok, not sure why I did it that way 🤣

> Should I remove the fallback from here and add a default to build.yaml?

Lets just leave it as is, I don't want to accidentally break something.